app/controllers/api/metrics_controller.rb

```diff
@@ -136,17 +138,7 @@ def build_metric
   end
 
   def find_plan_and_service
-    @plan = find_application_plan || find_end_user_plan
+    @plan = current_account.application_plans.find(params[:application_plan_id])
```

Comment:
@josemigallas Are you sure that the plan is mandatory for this controller as well? I'm not sure :thinking: And if so, you need to remove the conditional in the next line because if there is no plan, this line will raise an error ActiveRecord::RecordNotFound (that will be rescued and responded with a 404)

Reply:
Oops, yeah you're right